### PR TITLE
AUDIO: Detect and skip BFW chunks in WAV files

### PR DESCRIPTION
**Pull request by tag2015**
This completes the previous PR #5053 

Modern WAV files may contain extra chunks before the FMT header, as specified in the Broadcast Wave Format extension. These chunks must be skipped to properly read the header.

One example of such occurrence is the waves/seagull sound effect in the intro of Perfect Tides, and on the pier

### CODE REVIEW

File: audio/decoders/wave.cpp [hunk 1]
```diff
@@ -62,15 +75,22 @@ bool loadWAVFromStream(Common::SeekableReadStream &stream, int &size, int &rate,
...
-		stream.skip(junksize + (junksize % 2));
-		stream.read(buf, 4);
+	while (1) { // skip junk/bext... chunks
+		int i;
+		for (i = 0; (i < EXT_CHUNKS) && (memcmp(buf, extensionChunks[i], 4) != 0); i++)
+			;
+		if (i != EXT_CHUNKS) { // found a known chunk
+			uint32 chunkSize = stream.readUint32LE();
+			// skip junk/ext chunk (add 1 byte if odd)
+			stream.skip(chunkSize + (chunkSize % 2));
```
Comment (bluegr) [inline audio/decoders/wave.cpp:85]: Does this apply to all these chunks?
There's code above for skipping "fact" chunks, that doesn't have this rounding up. Should it also be included with these chunks?

Comment (tag2015) [reply audio/decoders/wave.cpp:85]: The chunk size should always be even, it's indicated in the specs for the JUNK chunk https://www.daubnet.com/en/file-format-riff; I couldn't find mention of this among the BWF specs though.
I used this tensorflow implementation as a reference and it does the rounding up for all chunks too
https://github.com/tensorflow/io/blob/189e5e1205d03b84d1b44c2e97cf9e97c5f910be/tensorflow_io/core/kernels/audio_video_wav_kernels.cc#L98

Comment (bluegr) [reply audio/decoders/wave.cpp:85]: Thanks for the reference and the explanation :)

**Comment by bluegr** [comment]
Thanks, this looks good to be merged, since it is based on an already working implementation

**Pull request merged**